src/lib/exceptions.ts

```diff
@@ -0,0 +1,50 @@
+import { kclManager } from 'lib/singletons'
+import { reloadModule, getModule } from 'lib/wasm_lib_wrapper'
+import toast from 'react-hot-toast'
+
+let initialized = false
+
+/**
+ * WASM/Rust runtime can panic and the original try/catch/finally blocks will not trigger
+ * on the await promise. The interface will killed. This means we need to catch the error at
+ * the global/DOM level. This will have to interface with whatever controlflow that needs to be picked up
+ * within the error branch in the typescript to cover the application state.
+ */
+export const initializeWindowExceptionHandler = () => {
+  if (window && !initialized) {
+    window.addEventListener('error', (event) => {
+      void (async () => {
+        if (matchImportExportErrorCrash(event.message)) {
+          // do global singleton cleanup
+          kclManager.executeAstCleanUp()
+          toast.error(
+            'You have hit a KCL execution bug! Put your KCL code in a github issue to help us resolve this bug.'
+          )
+          try {
+            await reloadModule()
+            await getModule().default()
+          } catch (e) {
+            console.error('Failed to initialize wasm_lib')
+            console.error(e)
+          }
+        }
+      })()
```

Comment:
This is missing the report rejection. When you fire off an async call like this and there's an exception in it, it disappears since nothing is awaiting it. I remember you running into this problem before. One way we addressed it is by explicitly catching anything we fire-and-forget.

```suggestion
      })().catch(reportRejection)
```

Reply:
Added, thanks for reminding me of this. I'll know for next time.